torch/csrc/api/include/torch/detail/any.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+#include <torch/csrc/utils/memory.h>
+#include <torch/csrc/utils/variadic.h>
+
+#include <ATen/Error.h>
+
+#include <memory>
+#include <typeinfo>
+#include <utility>
+
+namespace torch {
+class Any;
+namespace detail {
+template <typename T>
+Any make_any(T&& value);
+} // namespace detail
+} // namespace torch
+
+namespace torch {
+/// A simplified implementation of `std::any` or `boost::any` which stores a
+/// type erased object, whose concrete value can be retrieved at runtime by
+/// checking if the `typeid()` of a requested type matches the `typeid()` of the
+/// object stored. It is simplified in that it does not handle copying, as we do
+/// not require it for our use cases. Moves are sufficient.
+class Any {
+ public:
+  /// Default construction is disallowed. Thus our invariant: an `Any` contains
+  /// an object at all times, from its construction, to its destruction.
+  Any() = delete;
```

Comment:
This is another difference from C++17 `std::any`, which has a default constructor and an "empty" state.

Reply:
Yes, but this is where I like the fact that we can simplify our implementation of any